internal/pkg/admission/mutating-webhook.go

```diff
@@ -103,4 +111,15 @@ func (r *Reconciler) mutatingWebhookConfiguration(
 			},
 		},
 	}
+
+	// set policy as the owner of the webhook
+	scheme := runtime.NewScheme()
+	if err := policiesv1.AddToScheme(scheme); err != nil {
+		return nil, fmt.Errorf("cannot add policies.kubewarden.io/v1 to scheme: %w", err)
+	}
+	if err := controllerutil.SetControllerReference(policy, &webhook, scheme); err != nil {
```

Comment:
Should not we use `SetOwnerReference`? 
https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.16.3/pkg/controller/controllerutil#SetOwnerReference

Reply:
I had the same mistake when I first tried it :D. See the commit message of https://github.com/kubewarden/kubewarden-controller/pull/566/commits/db21d3cee95740bc79594fd1bf357085352dd823. Note that `SetOwnerReference` sets the OwnerReference field, but doesn't get the object reconciled, which is what we want here.

Reply:
Indeed. My fault. I skin read the docs and missed that. I was intrigued by method name. hehe